### PR TITLE
Add global price multipliers for all singlebuilt items

### LINKED ISSUE #1: Global singlebuilt item cost multiplier
1. players need a lot more items in CI than in vanilla
2. we want prototype armory to be balanced for vanilla
3. we want CIBridge mods to be balanced for CI

With those three facts in mind, I think the best solution is to balance all prices here for CI, but have a global cost multiplier. Here in PA, the default is 2 or 3 or whatever it needs to be, and in CI the multiplier is set to 1. Then we have an OPTC function that loops through all the singlebuildable items and multiplies their prices by this value.

### PR DESCRIPTION
**Pull request by cannonfodder1**
Closes #1 
Requires merge of https://github.com/WOTCStrategyOverhaul/CovertInfiltration/pull/524

### CODE REVIEW

File: PrototypeArmoury/Src/PrototypeArmoury/Classes/PATemplateMods.uc [hunk 1]
```diff
@@ -13,17 +13,22 @@ struct ItemCostOverride
 	var StrategyCost NewCost;
 };
 
+var config(StrategyTuning) array<StrategyCostScalar> ResourceCostScalars;
+var config(StrategyTuning) array<float> ArtifactCostMultiplier;
```
Comment (Xymanek) [inline PrototypeArmoury/Src/PrototypeArmoury/Classes/PATemplateMods.uc]: Why not use `StrategyCostScalar` for this? Just have one variable

File: PrototypeArmoury/Config/Items/XComStrategyTuning.ini [hunk 2]
```diff
@@ -276,3 +248,59 @@ Cost=(ResourceCosts[0]=(ItemTemplateName="Supplies", Quantity=30), ResourceCosts
 
 [PoweredTemplarArmor_Diff_3 X2ArmorTemplate]
 Cost=(ResourceCosts[0]=(ItemTemplateName="Supplies", Quantity=65), ResourceCosts[1]=(ItemTemplateName="AlienAlloy", Quantity=10), ResourceCosts[2]=(ItemTemplateName="EleriumDust", Quantity=7))
+
+; TODO: make this MCM toggled
+; Turn this option to true to change the cost of hardcoded vanilla items according to the arrItemCostOverrides defined below
+; The overridden prices generally make items cheaper for campaigns where you will need a lot of each item. Not recommended for a vanilla campaign.
+EnableUtilityItemOverrides=false
```
Comment (Xymanek) [inline PrototypeArmoury/Config/Items/XComStrategyTuning.ini]: Very much 👎 

* CI cannot override this
* You are disabling a system, while naming it by a specific use - what if some mod/bridge uses `arrItemCostOverrides` to change prices only when PA is loaded?

Comment (cannonfodder1) [reply PrototypeArmoury/Config/Items/XComStrategyTuning.ini]: In that case, we have two options:
- multiply ItemCostOverrides by the global price multiplier as well
- remove all the ItemCostOverride entries from PA except for a commented example and move them to CI

Comment (Xymanek) [reply PrototypeArmoury/Config/Items/XComStrategyTuning.ini]: The latter probably makes more sense. The utility items are individual build anyway, their cost re balancing was done for CI

File: PrototypeArmoury/Src/PrototypeArmoury/Classes/PAHelpers.uc [hunk 3]
```diff
@@ -2,14 +2,29 @@ class PAHelpers extends Object abstract;
...
 
-	DLCs = class'Helpers'.static.GetInstalledDLCNames();
+    EventManager = `ONLINEEVENTMGR;
+	
+	`PA_Trace("Installed Mods: " $ EventManager.GetNumDLC());
 
-	return DLCs.Find(DLCName) != INDEX_NONE;
+    for (Index = 0; Index < EventManager.GetNumDLC(); Index++)    
+    {
+		`PA_Trace("--> " $ EventManager.GetDLCNames(Index));
```
Comment (Xymanek) [inline PrototypeArmoury/Src/PrototypeArmoury/Classes/PAHelpers.uc]: Could you remove this trace? Will unnecessary spam logs with possibility the entire list of loaded mods every time a LDC check is made

File: PrototypeArmoury/Config/Items/XComStrategyTuning.ini [hunk 4]
```diff
@@ -24,6 +24,39 @@
 
 AutoBlackMarketPriceMult=0.5
 
+; These are the global cost multipliers for every individually-built item
+; Artifacts are stuff like corpses or elerium cores
+
+; This means that any price, defined anywhere, for any item in arrMakeItemBuildable will be multiplied by 2
+; With Covert Infiltration installed, the global multiplier is set to 1
```
Comment (Xymanek) [inline PrototypeArmoury/Config/Items/XComStrategyTuning.ini]:
```suggestion
; With Covert Infiltration installed, the global multiplier is set to 1 - the configs below are ignored!
```

File: PrototypeArmoury/Src/PrototypeArmoury/Classes/PAHelpers.uc [hunk 5]
```diff
@@ -2,14 +2,25 @@ class PAHelpers extends Object abstract;
...
 		return true;
 
-	DLCs = class'Helpers'.static.GetInstalledDLCNames();
-
-	return DLCs.Find(DLCName) != INDEX_NONE;
+    EventManager = `ONLINEEVENTMGR;
+	
+    for (Index = 0; Index < EventManager.GetNumDLC(); Index++)    
+    {
+        if(EventManager.GetDLCNames(Index) == name(DLCName))
```
Comment (Xymanek) [inline PrototypeArmoury/Src/PrototypeArmoury/Classes/PAHelpers.uc]:
```suggestion
        if (EventManager.GetDLCNames(Index) == DLCName)    
```
And change the function signature to `IsDLCLoaded (name DLCName)`

**Review by Xymanek — CHANGES_REQUESTED** [review]
Make the changes and merge

**Pull request merged**